pkg/apply/task/inv_set_task.go

```diff
@@ -54,79 +83,99 @@ func (i *InvSetTask) Identifiers() object.ObjMetadataSet {
+	klog.V(4).Infof("keep in inventory %d skipped applies", len(applySkips))
+	invObjs = invObjs.Union(applySkips)
+
+	// If an object failed to delete and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	// It's likely that all the delete failures are already in the inventory,
+	// because the set of resources to prune comes from the inventory,
+	// but we're doing the intersection anyway just to be sure.
+	pruneFailures := i.PrevInventory.Intersection(im.FailedDeletes())
+	klog.V(4).Infof("set inventory %d failed prunes", len(pruneFailures))
+	invObjs = invObjs.Union(pruneFailures)
+
+	// If an object skipped delete and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	// It's likely that all the skipped deletes are already in the inventory,
+	// because the set of resources to prune comes from the inventory,
+	// but we're doing the intersection anyway just to be sure.
+	pruneSkips := i.PrevInventory.Intersection(im.SkippedDeletes())
+	klog.V(4).Infof("keep in inventory %d skipped prunes", len(pruneSkips))
+	invObjs = invObjs.Union(pruneSkips)
+
+	// If an object is abandoned, then remove it from the inventory.
+	abandonedObjects := taskContext.AbandonedObjects()
+	klog.V(4).Infof("remove from inventory %d abandoned objects", len(abandonedObjects))
+	invObjs = invObjs.Diff(abandonedObjects)
+
+	// If an object is invalid and was previously stored in the inventory,
+	// then keep it in the inventory so it can be applied/pruned next time.
+	invalidObjects := i.PrevInventory.Intersection(taskContext.InvalidObjects())
+	klog.V(4).Infof("keep in inventory %d invalid objects", len(invalidObjects))
+	invObjs = invObjs.Union(invalidObjects)
+
+	klog.V(4).Infof("get the apply status for %d objects", len(invObjs))
+	objStatus := taskContext.InventoryManager().Inventory().Status.Objects
+
+	klog.V(4).Infof("set inventory %d total objects", len(invObjs))
+	err := i.InvClient.Replace(i.InvInfo, invObjs, objStatus, i.DryRun)
+
+	klog.V(2).Infof("inventory set task completing (name: %q)", i.TaskName)
+	return err
 }
 
-// Cancel is not supported by the InvSetTask.
-func (i *InvSetTask) Cancel(_ *taskrunner.TaskContext) {}
+// deleteInventory deletes the inventory object from the cluster.
+func (i *DeleteOrUpdateInvTask) deleteInventory() error {
+	klog.V(2).Infof("delete inventory task starting (name: %q)", i.Name())
+	err := i.InvClient.DeleteInventoryObj(i.InvInfo, i.DryRun)
+	// Not found is not error, since this means it was already deleted.
+	if apierrors.IsNotFound(err) {
+		err = nil
+	}
+	klog.V(2).Infof("delete inventory task completing (name: %q)", i.Name())
+	return err
+}
 
-// StatusUpdate is not supported by the InvSetTask.
-func (i *InvSetTask) StatusUpdate(_ *taskrunner.TaskContext, _ object.ObjMetadata) {}
+func (i *DeleteOrUpdateInvTask) success(taskContext *taskrunner.TaskContext) bool {
```

Comment:
func comment: success returns true when destroy actuation and reconciliation was fully successful. When true, it's safe to delete the inventory.

the name could maybe also be destroySuccessful, since it doesn't work for checking apply success.

Reply:
done